core-v1/src/main/scala/dsl/BuildExprDsl.scala

```diff
@@ -53,6 +53,64 @@ trait BuildExprDsl extends DebugExprDsl with WrapArityMethods {
 
   def none[O](implicit opO: OP[Option[W[O]]]): Any ~:> Option[W[O]] = Expr.Const(None: Option[W[O]])
 
+  def define[T](factType: FactType[T]): DefineBuilder[T]
+
+  abstract class DefineBuilder[T](factType: FactType[T]) {
+
+    def oneFrom(
+      defnExpr: Any ~:> W[T],
+    )(implicit
+      opWT: OP[W[T]],
+      opT: OP[T],
+      opF: OP[Seq[TypedFact[T]]],
+    ): Expr.Define[Any, Id, T, OP]
+
+    def from[C[_] : Functor : Foldable](
+      defnExpr: Any ~:> C[W[T]],
+    )(implicit
+      opWT: OP[W[T]],
+      opCT: OP[C[T]],
+      opT: OP[T],
+      opF: OP[Seq[TypedFact[T]]],
+    ): Expr.Define[Any, C, T, OP]
+
+    def fromInput[I, C[_] : Functor : Foldable](
+      buildDefnExpr: I =~:> C[W[T]],
+    )(implicit
+      opI: OP[I],
+      opWT: OP[W[T]],
+      opCT: OP[C[T]],
+      opT: OP[T],
+      opF: OP[Seq[TypedFact[T]]],
+    ): Expr.Define[I, C, T, OP]
+
+    final def oneFromConst(
+      value: T,
+    )(implicit
+      opCT: OP[T],
+      opF: OP[Seq[TypedFact[T]]],
+    ): Expr.Define[Any, Id, T, OP] =
+      Expr.Define(factType, Expr.Const(value): Any ~:> Id[T])
+
+    final def fromConst[C[_] : Foldable](
+      values: C[T],
+    )(implicit
+      opT: OP[Seq[T]],
+      opF: OP[Seq[TypedFact[T]]],
+    ): Expr.Define[Any, Seq, T, OP] =
+      Expr.Define(factType, Expr.Const(Foldable[C].toList(values): Seq[T]))
+  }
+
+  // TODO: Support function builder syntax
+  final def usingDefinitions[I, O](
+    definitions: Expr.Definition[I, OP]*,
+  )(
+    thenExpr: Any ~:> O,
+  )(implicit
+    opO: OP[O],
+  ): Expr.UsingDefinitions[I, O, OP] =
+    Expr.UsingDefinitions(definitions, thenExpr)
```

Comment:
Remove the TODO, since this is done. Also, maybe deprecate this method for backwards compatibility only?